copy.go

```diff
@@ -188,3 +190,126 @@ func (r *BinaryCopyReader) Read(ctx context.Context) (_ []any, err error) {
 
 	return row, nil
 }
+
+// read text data
+
+type TextCopyReader struct {
+	typeMap    *pgtype.Map
+	reader     *CopyReader
+	scanners   []Scanner
+	csvReader  *csv.Reader
+	buffer     *bytes.Buffer
+	bufScanner *bufio.Scanner
+	nullValue  string // PostgreSQL NULL value string (default empty)
+}
+
+func NewTextColumnReader(ctx context.Context, copy *CopyReader) (_ *TextCopyReader, err error) {
+	tm := TypeMap(ctx)
+	if tm == nil {
+		return nil, errors.New("postgres connection info has not been defined inside the given context")
+	}
+
+	scanners := make([]Scanner, len(copy.columns))
+	for index, column := range copy.columns {
+		scanners[index], err = NewScanner(tm, column, TextFormat)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	csvReaderBuffer := &bytes.Buffer{}
+	csvReader := csv.NewReader(csvReaderBuffer)
+	csvReader.Comma = ','
+	csvReader.TrimLeadingSpace = false
+	csvReader.LazyQuotes = true // Handle non-standard quoting
+
+	reader := &TextCopyReader{
+		typeMap:    tm,
+		reader:     copy,
+		scanners:   scanners,
+		csvReader:  csvReader,
+		buffer:     csvReaderBuffer,
+		bufScanner: bufio.NewScanner(csvReaderBuffer),
+		nullValue:  "", // Default NULL value is empty string
+	}
+
+	return reader, nil
+}
+
+// Read reads a single row from the copy-in stream. The read row is returned as a
+// slice of any values. If the end of the copy-in stream is reached, an io.EOF error
+// is returned.
+func (r *TextCopyReader) Read(ctx context.Context) (_ []any, err error) {
+	if ctx.Err() != nil {
+		return nil, ctx.Err()
+	}
+
+	for {
+		// Try to read a CSV record from the current buffer
+		record, err := r.csvReader.Read()
+		if err == io.EOF {
+			// CSV reader hit EOF, need more data from copy stream
+			err = r.reader.Read()
+			if err == io.EOF {
+				// End of copy stream, no more data
+				return nil, io.EOF
+			}
+			if err != nil {
+				return nil, err
+			}
+
+			// Process PostgreSQL CSV escape sequences before adding to buffer
+			processedData := r.preprocessPostgreSQLCSV(r.reader.Msg)
+			r.buffer.Write(processedData)
+
+			// Clear the message after copying to buffer
+			r.reader.Msg = r.reader.Msg[:0]
+
+			// Continue loop to try reading CSV record again
+			continue
+		} else if err != nil {
+			return nil, fmt.Errorf("CSV parsing error: %w", err)
+		}
```

Comment:
I would like to suggest to return this error using a guard clause instead.

```suggestion
		if err != nil && err != io.EOF {
			return nil, fmt.Errorf("CSV parsing error: %w", err)
		}
```